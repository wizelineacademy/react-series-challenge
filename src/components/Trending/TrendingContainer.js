import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import trendingGifsActions from '../../actions/trendingGifs';
import favoriteGifsActions from '../../actions/favoriteGifs';
import Trending from './Trending';

const mapStateToProps = (state) => {
  const {
    trendingGifs,
    favoriteGifs,
    filterText,
  } = state;
  return {
    trendingGifs,
    favoriteGifs,
    filterTrendingText: filterText.trendingFilter,
  };
};

const mapDispatchToProps = (dispatch) => {
  const { fetchTrending, searchTrending } = trendingGifsActions.creators;
  const { fetchFavorite } = favoriteGifsActions.creators;
  return bindActionCreators({
    fetchTrending,
    searchTrending,
    fetchFavorite,
  }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(Trending);
