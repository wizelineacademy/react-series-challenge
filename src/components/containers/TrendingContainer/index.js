import React, { Component } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import {
  actionTrendingGetDataRequest,
  actionSetToFav,
  actionSearchTermGetDataRequest
} from "../../../store/actions";
import { getTrendingData } from "../../../store/selectors/trendingSelectors";
import { getFavsDataIds } from "../../../store/selectors/favsSelectors";
import SearchBar from "../../molecules/SearchBar/index";
import GiPanel from "../../molecules/GiPanel/index";

export class TrendingContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchTerm: ""
    };
  }

  componentDidMount() {
    const { actionTrendingGetDataRequest } = this.props;
    if (actionTrendingGetDataRequest !== undefined) {
      actionTrendingGetDataRequest({});
    }
  }

  addToFavs = item => {
    this.props.actionSetToFav(item);
  };

  getInputValue = e => {
    this.props.actionSearchTermGetDataRequest(e.target.value);
  };

  render() {
    return [
      <SearchBar key={0} getInputValue={this.getInputValue} />,
      <GiPanel
        key={1}
        data={this.props.trending}
        toggleFavs={this.addToFavs}
        selectedFavs={this.props.favs}
      />
    ];
  }
}

function mapStateToProps(state) {
  let { trending, favs } = state;

  favs = getFavsDataIds(favs);
  trending = getTrendingData(trending);

  return {
    trending,
    favs
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      actionTrendingGetDataRequest,
      actionSetToFav,
      actionSearchTermGetDataRequest
    },
    dispatch
  );
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TrendingContainer);
