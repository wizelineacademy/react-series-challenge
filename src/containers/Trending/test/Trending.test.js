import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import TrendingContainer from '..';
import { createMockStore } from 'redux-test-utils';

const props = {
  isFavorite: true,
  gif: {
    title:'title',
    images: {
      downsized_large: {
        url: 'test'
      }
    }
  },
  searchGif: jest.fn(),
  setFavorite: jest.fn(),
  history: {
    goBack: jest.fn()
  },
  match: {
    params: {
      id: 1
    }
  },
  location: {
    pathname: '/test'
  }
}

describe('TrendingContainer', () => {
  it('should render correctly', () => {
    const output = shallow(
      <TrendingContainer.WrappedComponent {...props}/>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should be defined', () => {
   expect(TrendingContainer).toBeDefined();
  });
  it("should render successfully ", () => {
  const testState = {
    gifs: [],
    favoriteGifs: []
  };
  const store = createMockStore(testState)
  const component = shallow(<TrendingContainer store={store} />);
  expect(component).toBeDefined();
  });
});
