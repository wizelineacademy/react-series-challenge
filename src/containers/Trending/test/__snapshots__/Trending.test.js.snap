// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrendingContainer should render correctly 1`] = `
<styled.div>
  <withRouter(SearchBar) />
  <Menu
    location="/test"
  />
</styled.div>
`;
